### PR TITLE
fix(api): ensure exact delay floats are preserved in command log

### LINKED ISSUE #7066: bug(api): delay() command log truncates seconds float to an integer
# Overview

Delay() function rounds down anytime the seconds float has a decimal 

# Steps to reproduce

Create code with delay() function where either seconds had a decimal or where minutes had a decimal under 0.0167
Ran code in command window with opentrons simulate and on run app 

Example code used:
``` python
from opentrons import protocol_api
metadata = {'apiLevel': '2.7',
            'author': 'Alex '}
def run(protocol: protocol_api.ProtocolContext):
    
 tiprack_1 = protocol.load_labware('opentrons_96_tiprack_300ul',location = '1')
 p1000 = protocol.load_instrument('p300_single_gen2','left', tip_racks=[tiprack_1])
 p1000.starting_tip = tiprack_1.wells_by_name()['C1']
 p1000.pick_up_tip()
 protocol.delay(seconds = 2.4)
 p1000.drop_tip()
 p1000.pick_up_tip()
 p1000.drop_tip() 
```

# Current behavior

Displaying 2 seconds 

# Expected behavior

For delay (seconds = 2.4)
Delaying for 2.4 seconds

### PR DESCRIPTION
**Pull request by mcous**
## Overview

This PR fixes #7066. See ticket + thread for problem details.

## Changelog

- fix(api): ensure exact delay floats are preserved in command log

## Review requests

This PR adds unit tests that cover its changes, but to smoke test, run a protocol with `ctx.delay(seconds=0.5)` and confirm the run log says `0.5 seconds` and not `0 seconds`.

## Risk assessment

Low.

- Changes are for the run log only, which do not affect actual protocol behavior
- Changes are covered by unit tests

### CODE REVIEW

File: api/src/opentrons/commands/protocol_commands.py [hunk 1]
```diff
@@ -16,16 +16,17 @@ def comment(msg):
 
 def delay(seconds, minutes, msg=None):
     td = timedelta(minutes=minutes, seconds=seconds)
-    minutes, seconds = divmod(td.seconds, 60)
+    actual_min, whole_sec = divmod(td.seconds, 60)
```
Comment (amitlissack) [inline api/src/opentrons/commands/protocol_commands.py]: Do you think you could use `timedelta.total_seconds()` instead of `td.seconds`?

**Review by amitlissack — CHANGES_REQUESTED** [review]

**Review by amitlissack — CHANGES_REQUESTED** [review]
I think this could be simpler.

File: api/src/opentrons/commands/protocol_commands.py (hunk 1 above)
Comment (mcous) [reply api/src/opentrons/commands/protocol_commands.py]: Hah didn't see that method on timedelta at all...

**Review by amitlissack — APPROVED** [review]
Very nice

**Review by sfoster1 — APPROVED** [review]
Nice!

**Comment by mcous** [comment]
Gonna hold off on merging this one until #7074 is merged since this isn't dire and this PR will be easier to resolve any conflicts in

**Pull request merged**